### PR TITLE
Yaw drift compensation

### PR DESCRIPTION
**Pull request by Louka3000**
This calculates the yaw difference between resets and applies a counter-rotation.
Tested by several people on our Discord, which mostly say that it works great and allows playing much longer without resetting.
Has settings for compensation amount, global and individual (tracker) toggles as well as choosing how many past resets are used.

Protocol PR: https://github.com/SlimeVR/SolarXR-Protocol/pull/85

### CODE REVIEW

**Comment by ImUrX** [comment]
Blocked until PR gets merged and the submodule points to that commit

File: server/src/main/java/com/jme3/math/Quaternion.java
Comment (ButterscotchV) [inline server/src/main/java/com/jme3/math/Quaternion.java:950]: If the size is `0` then this will have a problem.

File: server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:485]:
```suggestion
				// Make it so recent Quaternions weigh more
```

File: server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:539]:
```suggestion
				// Make it so recent Quaternions weigh more
```

File: server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java
Comment (ButterscotchV) [inline server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:497]: Can you explain what's going on here? This is so difficult to understand

Comment (Louka3000) [reply server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:497]: it's making it so the most recent data in the list is worth more.
To be precise, if you have 10, 20 and 5, it'll do:
```
10 = 10 + (20 / 3)
20 = 20 - (20 / 3)
```
and repeat for each one up to the last.
```
13.33 = 13.33 + (5 / 3)
5 = 5 - (5 / 3)
```
`13.33` being the second element in the list (used to be 20)
`3` being the amount of numbers of the list.

Comment (ButterscotchV) [reply server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:506]: ah hm I see, is there any way you could write a comment explaining this a little better in the code?

Comment (Louka3000) [reply server/src/main/java/dev/slimevr/vr/trackers/IMUTracker.java:506]: Sure

File: server/src/main/java/com/jme3/math/Quaternion.java
Comment (ButterscotchV) [inline server/src/main/java/com/jme3/math/Quaternion.java:937]: Is this supposed to be here?
```suggestion
			throw new IllegalArgumentException("qn's length can't be 0");
```

File: server/src/main/java/com/jme3/math/Quaternion.java
Comment (ButterscotchV) [inline server/src/main/java/com/jme3/math/Quaternion.java:970]: Same as my last comment
```suggestion
			throw new IllegalArgumentException("qn's length can't be 0");
```

File: server/src/main/java/com/jme3/math/Quaternion.java
Comment (Louka3000) [reply server/src/main/java/com/jme3/math/Quaternion.java:937]: Oops, I blame IDEA for putting 2 ' when I put one...

**Review by ButterscotchV — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Comment by Eirenliel** [comment]
pwease webase 🥺

**Comment by Louka3000** [comment]
Done @Eirenliel

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**
